### DISCUSSION TITLE
Doku request

### DISCUSSION THREAD

**Discussion by stxShadow** [shining-man/bsc_fw#3] [General]
Moin Florian ! 

vielen Dank für dieses Projekt. Ich betreibe den YT Kanaln "Meine Energiewende" und in unserem Discord Chat sind eine Leute wirklich "heiss" auf Dein Projekt. -> NEEY Balancer abzufragen ist bisher leider nur ein Traum.

Daher würde ich gerne im Namen des Kanals um etwas Doku bitten. Wir hatten auch schon auf "meintechblog.de" geschrieben. Du erreichst mich unter jens@solwatch.eu.

Viele Grüße
Jens

**Comment by shining-man** [comment]
Nicht Florian, sondern Tobias;-)
Ich habe mit der Doku begonnen und werde sie nach und nach erweitern. 
Die Doku ist im Wiki.

**Comment by dersch81** [comment]
Dein Projekt macht ziemlich Lust es einzusetzen. Ich wollte das Meiste tatsächlich über den Pi auf dem VenusOS mit NodeRed abbilden. Aber mit deinem Projekt macht das dann etwas mehr Spaß! Wo bekomme ich deine Draft Platine her? Würdest du mal eine Batch davon herstellen lassen? 

Ich habe hier die JBD als auch die NEEY rumliegen und fange bald an meine zwei 16s Pakete als auch das ESS zu bauen. 

Da VenusOS sich nur eines der über USB angebundenen BMS als Master nimmt und damit die Statistik nicht stimmt weil das zweite, dritte, vierte, etc BMS nicht beachtet wird suche ich noch eine Lösung Venus die Daten aufbereitet zur Verfügung zu stellen. Mit deiner Lösung wäre das wahrscheinlich am besten ohne dabei über weitere Middleware zu gehen. Aber bezgl des WIE habe ich noch einige Fragezeichen. Also wir brauchen DOKU, MEHR DOKU :) 

PS: würdest du noch die Disskusionen hier freischalten? Dann muss nicht extra ein Issue aufgemacht werden um was zu Fragen oder Vorzustellen ;) 

PSS: Nutzt du zum übermitteln der Daten an Venus evtl https://github.com/freakent/dbus-mqtt-devices?

**Reply by shining-man** [reply]
Ich habe von den Leiterplatten noch 2-3 über, die ich weitergeben kann. Die Bauteile sollten fast alle einfach beschaffbar sein (z.B. Reichelt). Hier könnte ich noch eine Stückliste erstellen.

Das Addieren bzw. verrechnen der Daten von mehreren BMSen ist kein Problem. Kann ich bei bedarf einbauen. 
Ich selber nutze keinen Victron WR. Bei mir läuft die Kommunikation zum WR über den CAN-Bus. Dies sollte im Prinzip auch bei Victron (Cerbo) gehen. Ob es mit VenusOS auf einem RaspberryPi funktioniert weiß ich nicht. Es gibt für den Raspi CAN Shields. Nur ob das dann funktioniert weiß ich nicht. Vielleicht kennt sich hier der Victron Spezialist Jens aus?
Eine andere Möglichkeit wäre es über die serielle Schnittstelle zum Venus zu senden. Dies müsste ich aber erst noch einbauen.

An der Doku bin ich dran. Gebt mir ein paar Tipps in welchem Bereich die Doku am wichtigsten ist:
- Bestimmte Funktionen?
- Hardware?
- Schnittstellen?
- Interner Aufbau der Software?

**Reply by dersch81** [reply]
Ich würde gerne eine Leiterplatte nehmen. Das mit dem CAN BUS auf einem RaspiShield kann ich gerne austesten denn das habe ich auf dem Pi auf dem Venus läuft. Also wenn du das Addieren der BMS Daten mit einbauen kannst wäre das sehr geil über diesem Weg Venus die Daten zu liefern.

Zunächst würde ich für die Doku wichtig finden wie man es korrekt einsetzt. Als Use Cases welche auch die Funktionen beschreibt. Ich mache gerne mit um einen weiteren UseCase zu zeigen.

**Reply by shining-man** [reply]
Wenn du mir eine Mail mit deiner Adresse schreibst, dann sende ich dir eine Leiterplatte zu.

**Comment by shining-man** [comment]
Ich habe im Wiki eine weitere Seite angelegt und werde dort nach und nach Use Cases beschreiben.
Es kann gerne eine Rückmeldung kommen, ob dies hilfreich ist.
